frontend/src/lib/arrowProto.ts

```diff
@@ -15,16 +15,14 @@
  * limitations under the License.
  */
 
-// (HK) TODO: Everything here must be heavily refactored.
-// Do NOT use this in production.
-
 import { Column, Table } from "apache-arrow"
-import { isEqual, range, unzip } from "lodash"
+import { range, unzip } from "lodash"
+import { Data, Index, IndexType, Quiver } from "src/lib/Quiver"
 
-export interface DataTable {
-  index: any
-  columns: any[][]
-  data: any[][]
+export interface DataFrame {
+  index: Index
+  columns: string[][]
+  data: Data
```

Comment:
Docstrings plz :)

Also, why are some of these types defined in `arrowProto`, and others are in `Quiver`?

Reply:
Merged `arrowProto` and `Quiver`